libraries/chain/db_maint.cpp

```diff
@@ -800,6 +800,83 @@ void update_and_match_call_orders( database& db )
+ * feeds. This method will run at the hardfork time, and erase (or nullify) feeds
+ * that have incorrect backing assets.
+ *
+ * @param db the database
+ */
+void cleanup_invalid_feeds_hf_868( database& db )
+{
+   // for each market issued asset
+   const auto& asset_idx = db.get_index_type<asset_index>().indices().get<by_type>();
+   for(auto asset_itr = asset_idx.lower_bound(true); asset_itr != asset_idx.end(); ++asset_itr)
+   {
+      const auto& current_asset = *asset_itr;
+      // Incorrect witness & committee feeds can simply be removed.
+      // For non-witness-fed and non-committee-fed assets, set incorrect
+      // feeds to price(), since we can't simply remove them. For more information:
+      // https://github.com/bitshares/bitshares-core/pull/832#issuecomment-384112633
+      bool is_witness_or_committee_fed = false;
+      if ( current_asset.options.flags & (witness_fed_asset | committee_fed_asset) )
+         is_witness_or_committee_fed = true;
+
+      // for each feed
+      const asset_bitasset_data_object& bitasset_data = current_asset.bitasset_data(db);
+      // NOTE: If HF343 and HF868 roll out at the same time
+      // you can remove this double, and the check_call_orders()
+      // below that relies on it. HF343 will take care of the issue
+      // and there is no need for the extra check_call_orders() call
+      auto old_price = bitasset_data.settlement_price;
+      bool feeds_changed = false; // did any feed change
+      auto itr = bitasset_data.feeds.begin();
+      while (itr != bitasset_data.feeds.end())
+      {
+         // If the feed is invalid
+         if ( (*itr).second.second.settlement_price.quote.asset_id != bitasset_data.options.short_backing_asset
+               && (is_witness_or_committee_fed || (*itr).second.second.settlement_price != price() ) ) {
+            feeds_changed = true;
+            db.modify(bitasset_data, [&itr, is_witness_or_committee_fed](asset_bitasset_data_object& obj) {
+               if (is_witness_or_committee_fed)
+               {
+                  // erase the invalid feed
+                  itr = obj.feeds.erase(itr);
+               }
+               else
+               {
+                  // nullify the invalid feed
+                  (*obj.feeds.find( (*itr).first )).second.second.settlement_price = price();
+                  ++itr;
+               }
+            });
+         }
+         else
+         {
+            // Feed is valid. Skip it.
+            ++itr;
+         }
+      } // end loop of each feed
+
+      // if the feeds were modified, update the median feed
+      if (feeds_changed) {
+         wlog("Found invalid feed for asset ${asset_id} during hardfork core-868", ("asset_id", current_asset));
+         db.modify(bitasset_data, [&db](asset_bitasset_data_object &obj) {
+            obj.update_median_feeds( db.head_block_time() );
+         });
+      }
+
+      if ( HARDFORK_CORE_343_TIME != HARDFORK_CORE_868_TIME
```

Comment:
Please pass in the variable `to_update_and_match_call_orders` which was initialized in `perform_chain_maintenance()` for the "2 hard forks occur at same time" check.

Reply:
to_update_and_match_call_orders passed